org.eclipse.eclemma.ui.test/src/org/eclipse/eclemma/ui/DumpExecutionDataTest.java

```diff
@@ -0,0 +1,144 @@
+ *
+ * Contributors:
+ *    Evgeny Mandrikov - initial API and implementation
+ *
+ ******************************************************************************/
+package org.eclipse.eclemma.ui;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.eclipse.core.runtime.ILogListener;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.debug.core.DebugPlugin;
+import org.eclipse.debug.core.ILaunch;
+import org.eclipse.debug.core.ILaunchConfiguration;
+import org.eclipse.debug.core.ILaunchConfigurationType;
+import org.eclipse.debug.core.ILaunchConfigurationWorkingCopy;
+import org.eclipse.eclemma.core.CoverageTools;
+import org.eclipse.eclemma.core.ISessionManager;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.IPackageFragment;
+import org.eclipse.jdt.core.IPackageFragmentRoot;
+import org.eclipse.jdt.launching.IJavaLaunchConfigurationConstants;
+import org.eclipse.swtbot.eclipse.finder.SWTWorkbenchBot;
+import org.eclipse.swtbot.eclipse.finder.widgets.SWTBotView;
+import org.eclipse.swtbot.swt.finder.finders.UIThreadRunnable;
+import org.eclipse.swtbot.swt.finder.results.VoidResult;
+import org.eclipse.ui.PartInitException;
+import org.eclipse.ui.PlatformUI;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class DumpExecutionDataTest {
+
+	private static final SWTWorkbenchBot bot = new SWTWorkbenchBot();
+
+	private static ILaunch launch1;
+	private static ILaunch launch2;
+
+	@AfterClass
+	public static void resetWorkbench() {
+		bot.resetWorkbench();
+	}
+
+	@BeforeClass
+	public static void setup() throws Exception {
+		openCoverageView();
+
+		JavaProjectKit project = new JavaProjectKit();
+		project.enableJava5();
+		final IPackageFragmentRoot root = project.createSourceFolder();
+		final IPackageFragment fragment = project.createPackage(root, "example");
+		project.createCompilationUnit(fragment, "Example.java",
+				"package example; class Example { public static void main(String[] args) throws Exception { Thread.sleep(30000); } }");
+
+		JavaProjectKit.waitForBuild();
+
+		ILaunchConfiguration launchConfiguration = createLaunchConfiguration(project.javaProject, "example.Example");
+
+		launch1 = launchConfiguration.launch(CoverageTools.LAUNCH_MODE, null);
+		launch2 = launchConfiguration.launch(CoverageTools.LAUNCH_MODE, null);
+	}
+
+	@Test
+	public void test() throws Exception {
+		final LogListener logListener = new LogListener();
+		Platform.addLogListener(logListener);
+
+		SWTBotView coverageView = bot.viewByTitle("Coverage");
+		coverageView.show();
+
+		coverageView.toolbarButton("Dump Execution Data").click();
+		bot.shell("Dump Execution Data").activate();
+		bot.button("Cancel").click();
+
+		coverageView.toolbarButton("Dump Execution Data").click();
+		bot.shell("Dump Execution Data").activate();
+		bot.button("OK").click();
```

Comment:
@Godin Isn't that a bug of `ListDialog`? JavaDoc says:

    A dialog that prompts for one element out of a list of elements.

So expect the OK button not to be active when no item is selected.